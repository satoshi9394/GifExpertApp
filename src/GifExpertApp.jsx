import React, { useState } from 'react'
//components
import AddCategory from './components/AddCategory';
import GifGrid from './components/GifGrid';

const GifExpertApp = ({ defaultCategories = [] }) => {

  const [categories, setCategories] = useState(defaultCategories);


  // const handleAdd = () => {
    // const newSerie = 'HunterX';
    // setCategories([ newSerie, ...categories ]);
    // setCategories( cats => [ newSerie, ...cats ]);
  // }

  return (
    <div>
      <h2>GifExpertApp</h2>
      <AddCategory setCategories={ setCategories }/>
      <hr/>
      <ol>
        {
          categories.map( category => 
            <GifGrid 
              category={category} 
              key={category}
            />
          )
        }
      </ol>
    </div>
  );
}

export default GifExpertApp
