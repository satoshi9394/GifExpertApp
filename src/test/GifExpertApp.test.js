import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifExpertApp from '../GifExpertApp';


describe('Pruebas en <GifExpertApp />', () => {

  test('debe mostrar el componente correctamente ', () => {
    const wrapper = shallow( <GifExpertApp /> );
    expect( wrapper ).toMatchSnapshot();
  });
  test('debe demostrar una lista de categorias', () => {
    const categories = ['One punch', 'Dragon ball'];
    const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );
    expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('GifGrid').length ).toBe( categories.length );


  });
})