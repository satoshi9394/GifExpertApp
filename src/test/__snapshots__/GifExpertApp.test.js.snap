// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> debe demostrar una lista de categorias 1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoris
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One punch"
      key="One punch"
    />
    <GifGrid
      category="Dragon ball"
      key="Dragon ball"
    />
  </ol>
</div>
`;

exports[`Pruebas en <GifExpertApp /> debe mostrar el componente correctamente  1`] = `
<div>
  <h2>
    GifExpertApp
  </h2>
  <AddCategoris
    setCategories={[Function]}
  />
  <hr />
  <ol />
</div>
`;
